src/blueprints/Do.Blueprints.Service.Application/Core/Dotnet/DotnetCoreExtensions.cs

```diff
@@ -1,9 +1,52 @@
+using System.Reflection;
 using Do.Core;
 using Do.Core.Dotnet;
+using Do.Testing;
+using Shouldly;
 
 namespace Do;
 
 public static class DotnetCoreExtensions
 {
     public static DotnetCoreFeature Dotnet(this CoreConfigurator _) => new();
+
+    public static PropertyInfo? PropertyOf<T>(this Stubber _, string name) =>
+        typeof(T).GetProperty(name, BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance);
+
+    public static void ShouldBe<T>(this Type source) => source.ShouldBe(typeof(T));
+
+    public static void ShouldBeAbstract(this PropertyInfo source)
+    {
+        var getMethod = source.GetGetMethod(true);
+
+        getMethod.ShouldNotBeNull();
+        getMethod.ShouldBeAbstract();
+    }
+
+    public static void ShouldBeVirtual(this PropertyInfo source)
+    {
+        var getMethod = source.GetGetMethod(true);
+
+        getMethod.ShouldNotBeNull();
+        getMethod.ShouldBeVirtual();
+    }
+
+    public static MethodInfo? MethodOf<T>(this Stubber _, string name) =>
+        typeof(T).GetMethod(name, BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance);
+
+    public static void ShouldBeAbstract(this MethodInfo source)
+    {
+        source.IsAbstract.ShouldBeTrue();
+    }
+
+    public static void ShouldBeVirtual(this MethodInfo source)
+    {
+        source.IsVirtual.ShouldBeTrue();
+    }
+
+    public static void ShouldHaveOneParameter<T>(this MethodInfo source)
+    {
+        source.GetParameters().Length.ShouldBe(1);
+        source.GetParameters().First().ParameterType.ShouldBe<T>();
+    }

```

Comment:
why are these not in `CoreExtensions` place these there as well

Reply:
Reflection seems to me to be directly related to dotnet. It seems like we cannot generalize for core such as string, integer.